src/main/java/domain/LottoService.java

```diff
@@ -16,29 +16,42 @@ public LottoService() {
     }
 
     public LottoPrice inputPrice() {
-        LottoPrice lottoPrice = new LottoPrice(inputView.inputPurchasePrice());
-        resultView.printLottoCount(lottoPrice.lottoCount());
-        return lottoPrice;
+        return new LottoPrice(inputView.inputPurchasePrice());
     }
 
     public LottoTickets purchaseLotto(LottoPrice lottoPrice) {
         LottoTickets lottoTickets = new LottoTickets();
-        lottoTickets.readyLottoTicket(lottoPrice.lottoCount());
-        showAllLottoTickets(lottoTickets);
+        int manualCount = purchaseManualLotto(lottoTickets);
+        int autoCount = lottoPrice.lottoCount() - manualCount;
+        purchaseAutoLotto(lottoTickets, autoCount);
+        resultView.printLottoCount(manualCount, autoCount);
         return lottoTickets;
     }
 
+    public int purchaseManualLotto(LottoTickets lottoTickets) {
+        int count = inputView.inputManualLottoCount();
+        inputView.inputManualLottoNumber();
+        for (int i = 0; i < count; i++) {
+            lottoTickets.addLottoTicket(Lotto.create(inputView.inputLotto()));
+        }
+        return count;
+    }
+
+    public void purchaseAutoLotto(LottoTickets lottoTickets, int count) {
+        lottoTickets.readyLottoTicket(count);
+    }
```

Comment:
```suggestion
    public LottoTickets purchase(LottoPrice lottoPrice) {
        int manualCount = inputView.inputManualLottoCount();
        int autoCount = lottoPrice.autoCount(manualCount);

        LottoTickets manual = LottoTickets.manual(inputManual(manualCount));
        LottoTickets auto = LottoTickets.auto(autoCount);

        resultView.printLottoCount(manualCount, autoCount);
        return manual.add(auto);
    }
```

이 코드기반으로 한번 리팩토링해보시겠어요?

Reply:
감사합니다! 👍